Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -50,13 +51,27 @@ directCallStatusCallback callSid dialCallStatus recordingUrl_ callDuration = do
   runTransaction $ QCallStatus.updateCallStatus callStatus.id (exotelStatusToInterfaceStatus dialCallStatus) (fromMaybe 0 callDuration) recordingUrl
   return Ack
 
-getCustomerMobileNumber :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
-getCustomerMobileNumber callSid callFrom_ callTo_ callStatus = do
+makeEnteredDtmfFlowKey :: Text -> Text
+makeEnteredDtmfFlowKey callSid = "DriverEnteredDtmfFlow:CallSid-" <> callSid
+
+getCustomerMobileNumber :: (Redis.HedisFlow m r, CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
+getCustomerMobileNumber callSid callFrom_ callTo_ digits callStatus = do
   let callFrom = dropFirstZero callFrom_
   let callTo = dropFirstZero callTo_
   mobileNumberHash <- getDbHash callFrom
+  dtmfMobileHash <- getDbHash digits
   exophone <- CQExophone.findByPhone callTo >>= fromMaybeM (ExophoneDoesNotExist callTo)
-  driver <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash exophone.merchantId >>= fromMaybeM (PersonWithPhoneNotFound callFrom)
+  let enteredDtmfFlow = False
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  let enteredDtmfFlowKey = makeEnteredDtmfFlowKey callSid
+  Redis.setExp enteredDtmfFlowKey enteredDtmfFlow expTime
+  checkMobNo <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash exophone.merchantId
+  driver <- case checkMobNo of
+    Nothing -> do
+      person <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" dtmfMobileHash exophone.merchantId >>= fromMaybeM (PersonWithPhoneNotFound digits)
```

Comment:
you can calculate the hash of `digit` (dtmfMobileHash) in Nothing case instead of always ?